### PR TITLE
Allow IDN domain names in annotations 

### LINKED ISSUE #16783: 🐛 Bug Report: Entity Annotation invalid
### 📜 Description

When setting up the gitlab integration with organization catalog sync, no entities are created. Both users and groups show entity errors in the logs. Supposedly, the annotation values are invalid as the prefix is invalid. However, the annotation values `repo.xn--2kr066avsra.com/<something>` follow the specification.

### 👍 Expected behavior

properly sync the organization structure

### 👎 Actual Behavior with Screenshots

**User Log Example:**

```json
{
   "entity":"user:default/dragos.jurca",
   "level":"warn",
   "location":"url:repo.xn--2kr066avsra.com/dragos.jurca",
   "message":"Policy check failed for user:default/pit.wegner; caused by Error: \"annotations.repo.xn--2kr066avsra.com/user-login\" is not valid; expected a valid prefix and/or suffix but found \"repo.xn--2kr066avsra.com/user-login\". To learn more about catalog file format, visit: https://github.com/backstage/backstage/blob/master/docs/architecture-decisions/adr002-default-catalog-file-format.md",
   "plugin":"catalog",
   "service":"backstage",
   "type":"plugin"
}
```

**Group Log Example:**

```json
{
   "entity":"group:default/it-service",
   "level":"warn",
   "location":"url:repo.xn--2kr066avsra.com/teams/it-service",
   "message":"Policy check failed for group:default/it-service; caused by Error: \"annotations.repo.xn--2kr066avsra.com/team-path\" is not valid; expected a valid prefix and/or suffix but found \"repo.xn--2kr066avsra.com/team-path\". To learn more about catalog file format, visit: https://github.com/backstage/backstage/blob/master/docs/architecture-decisions/adr002-default-catalog-file-format.md",
   "plugin":"catalog",
   "service":"backstage",
   "type":"plugin"
}
```

### 👟 Reproduction steps

1. Add Gitlab Integration with self hosted Gitlab:
    ```yaml
    integrations:
      gitlab:
        - host: repo.xn--2kr066avsra.com
          token: ${GITLAB_TOKEN}
          apiBaseUrl: https://repo.xn--2kr066avsra.com/api/v4
    ```
3. Enable catalog provider for gitlab:
    ```yaml
    catalog:
      providers:
        gitlab:
          mygitlab:
            host: repo.xn--2kr066avsra.com
            orgEnabled: true
    ```
4. Modify `packages/backend/src/plugins/catalog.ts` to add `GitlabOrgDiscoveryEntityProvider` as an EntityProvider:
    ```js
    builder.addEntityProvider(
      GitlabOrgDiscoveryEntityProvider.fromConfig(env.config, {
        logger: env.logger,
          schedule: env.scheduler.createScheduledTaskRunner({
            frequency: { hours: 1 },
            timeout: { minutes: 50 },
            initialDelay: { seconds: 15 }
          }),
        }),
      );
    ```

### 📃 Provide the context for the Bug.

This seems to be an issue with backstage, not the gitlab provider.

### 🖥️ Your Environment

docker compose setup with all in one Dockerfile using backstage version `1.11.0`.

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by pitwegner**
## Hey, I just made a Pull Request!

Closes #16783 

I changed the regex for annotation key validation to support IDN domain names (and more generally, follow the [Kubernetes standards](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-label-names)).

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Tests for new functionality and regression tests for bug fixes
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/catalog-model/src/validation/CommonValidatorFunctions.ts [hunk 1]
```diff
@@ -91,7 +91,7 @@ export class CommonValidatorFunctions {
       typeof value === 'string' &&
       value.length >= 1 &&
       value.length <= 63 &&
-      /^[a-z0-9]+(\-[a-z0-9]+)*$/.test(value)
+      /^(?!\-)[a-z0-9\-]*[a-z0-9]$/.test(value)
```
Comment (Rugvip) [inline packages/catalog-model/src/validation/CommonValidatorFunctions.ts:94]:
```suggestion
      /^[a-z0-9]+(?:\-+[a-z0-9]+)*$/.test(value)
```

Could we keep the existing structure? prefer that over the negative lookahead

**Review by Rugvip — CHANGES_REQUESTED** [review]
Thank you! 👍

File: packages/catalog-model/src/validation/CommonValidatorFunctions.ts (hunk 1 above)
Comment (pitwegner) [reply packages/catalog-model/src/validation/CommonValidatorFunctions.ts:94]: Sure

**Comment by pitwegner** [comment]
anything missing for the merge?

**Review by Rugvip — APPROVED** [review]
Just time, sorry 😅 

Looks good! Let's :shipit:

**Pull request merged**